DotNet/RoadmapLogic/RoadmapImage.cs

```diff
@@ -212,5 +229,34 @@ private static void DrawShortLine(Image<Rgba32> image, Settings settings)
             new PointF[] { new PointF(settings.Margin.Left - 20, settings.Margin.Top - 25),
                 new PointF(settings.Margin.Left + 80, settings.Margin.Top - 25)});
         }
+
+        private static string BuildMessage(bool hasMissingProjects, bool hasProjectBeforeStartQuarter, bool hasProjectAfterLastQuarter)
```

Comment:
It might take a little design, but what about dealing with an array of error messages instead of these individual booleans? If you passed in an array of messages, this method could just concatenate the non-empty ones and return them. Or pass in a Dictionary<bool, string> so you know which error conditions should be reported.

Reply:
Passing an array of messages still requires to check for conditions to build an array of messages. Passing dicitionary <bool, string> is not any different from what the current code is doing; instead of getting the value from dictionary it's directly using the private constant variable.

Reply:
Well, it is a little different. The signature for the BuildMessage method would be

`BuildMessage(IDictionary<bool, string> messages)`

and that would never change. With your current design, you would have to keep adding more bool arguments whenever you wanted to handle a new type of error message:

`BuildMessage(bool hasMissingProjects, bool hasProjectBeforeStartQuarter, bool hasProjectAfterLastQuarter, bool hasSomeOtherError, hasYetAnotherError, ...)`

A lot of boolean function arguments probably means the function needs to be refactored to accept a composition object.